execution/process.go

```diff
@@ -261,5 +303,87 @@ func (p *ProcessByKey) ReadyForMore(ctx context.Context, tx storage.StateTransac
 }
 
 func (p *ProcessByKey) Close(ctx context.Context, storage storage.Storage) error {
+	p.garbageCollectorCtxCancel()
+	err := <-p.garbageCollectorCloseErrChan
+	if err == context.Canceled || err == context.DeadlineExceeded {
+	} else if err != nil {
+		return errors.Wrap(err, "couldn't stop garbage collector")
+	}
```

Comment:
Shouldn't this close function also drop all local data? (DropPrefix?)

Reply:
It should but only on storages used in methods `GetWatermark` and `Next` as rest of methods are called with engine prefixed tx. Will fix this

Reply:
Actually, now `Next` must be cleared separately because `StreamPrinter` calls `Next` without prefixing it...

Reply:
but storage used in GarbageCollector doesn't have to be cleared as it's creating transaction from prefixed (with pull engine streamID) storage, so pull engine will clear it